### PR TITLE
Extending TokenRequestContext to accept additional Claims

### PR DESCRIPTION
**Pull request by schaabs**
This PR adds a `Claims` property to `TokenRequestContext` to enable service clients to request additional claims when acquiring a token, as well as handle claims challenges from the service.

### CODE REVIEW

File: sdk/core/Azure.Core/api/Azure.Core.net461.cs [hunk 1]
```diff
@@ -417,6 +417,8 @@ public abstract partial class TokenCredential
         private readonly object _dummy;
         private readonly int _dummyPrimitive;
         public TokenRequestContext(string[] scopes, string? parentRequestId = null) { throw null; }
+        public TokenRequestContext(string[] scopes, string? parentRequestId = null, string? claims = null) { throw null; }
+        public string? Claims { get { throw null; } }
```
Comment (pakrym) [inline sdk/core/Azure.Core/api/Azure.Core.net461.cs:421]: "Claims" is plural but not a collection while `Scopes` is. Why?

Comment (schaabs) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:421]: The value of this property is actually an id token which is a complex json object containing a collection of claims which need to be included in the issued token (I will add more detail on this to the xml comments). Given that do you still have an issue with the property name being plural?

Comment (pakrym) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:421]: Hm, does it follow some spec?

Comment (schaabs) [reply sdk/core/Azure.Core/api/Azure.Core.net461.cs:421]: https://openid.net/specs/openid-connect-core-1_0-final.html#ClaimsParameter, I can link to this directly in the doc comments.

**Review by KrzysztofCwalina — APPROVED** [review]

**Review by christothes — APPROVED** [review]

**Pull request merged**